src/index.js

```diff
@@ -1,34 +1,76 @@
 import './style.css';
 
-const todoList = [
-  {
-    description: 'wake up early',
-    completed: false,
-    index: 0,
-  },
-  {
-    description: 'setup for morning session',
-    completed: false,
-    index: 1,
-  },
-  {
-    description: 'Coding with microverse',
-    completed: true,
-    index: 2,
-  },
-];
-
-const displayTodos = (todo) => {
-  const ul = document.querySelector('ul');
-  let content = '';
-  todo.forEach((tod) => {
-    content += `<li class="item">
-        <input type="checkbox" />
-        <p contenteditable="true">${tod.description}</p>
-      </li>`;
+class Todo {
+  constructor(desc, completed, id) {
+    this.desc = desc;
+    this.completed = completed;
+    this.id = id;
+  }
+}
+class UI {
+  static displayTodos = (todos) => {
+    const ul = document.querySelector('ul');
+    if (!todos.length) ul.innerHTML = '';
+    let content = '';
+    todos.forEach((tod) => {
+      content += `
+        <li id=${tod.id} class="item">
+          <div class='group'>
+            <input class='bdan' name='checkbox' type="checkbox" />
+            <p contenteditable="true">${tod.desc}</p>
+          </div>
+          <i class="fa-solid fa-trash"></i>
+        </li>`;
+    });
     ul.innerHTML = content;
-  });
+    content = '';
+  };
 
-  content = '';
-};
-displayTodos(todoList);
\ No newline at end of file
+  static generateId = () => Math.floor((1 + Math.random()) * 0x10000)
+    .toString(16)
+    .substring(1);
+
+  static clearValue = () => {
+    document.querySelector('.input').value = '';
+  };
+}
+
+class Store {
+  static getTodos = () => {
+    let todos;
+    if (!localStorage.getItem('todos')) todos = [];
+    else todos = JSON.parse(localStorage.getItem('todos'));
+    return todos;
+  };
+
+  static addTodo = (todo) => {
+    const todos = Store.getTodos();
+    todos.push(todo);
+    localStorage.setItem('todos', JSON.stringify(todos));
+    UI.displayTodos(todos);
+  };
+
+  static removeTodo = (id) => {
+    let todos = Store.getTodos();
+    todos = todos.filter((todo) => todo.id !== id);
+    localStorage.setItem('todos', JSON.stringify(todos));
+    UI.displayTodos(todos);
+  };
```

Comment:
Kindly make sure to update all indexes of the items after the one that was removed as this was clearly mentioned in the requirements => "Deleting a task should update all remaining items' indexes, so they represent the current list order and are unique(i.e. if you're deleting the first task index 1 from the list, the index of the next task(2) should set to 1).."

Reply:
@souad988 According to the requirement updated the issue and resolved the conflicts